.github/workflows/ci.yml

```
@@ -25,7 +25,16 @@ jobs:
 
       - name: Run lint
         run: |
-          make lint
+          # Ensure linting fails the build if issues are found
+          make lint || exit 1  
+
+      - name: Auto-format code
+        run: |
+          make format
```

I'd prefer to not create an automated commit just for formatting and simply check for it:
```suggestion
          make check-format || exit 1
```

---

Why not just:

"make check-format"

"|| exit 1" will just hide the original exit code. If it's false, it should fail anyway (false is non-zero in bash)